x/slashing/keys.go

```diff
@@ -34,7 +34,8 @@ func GetValidatorSlashingPeriodPrefix(v sdk.ConsAddress) []byte {
 // stored by *Tendermint* address (not operator address) followed by start height
 func GetValidatorSlashingPeriodKey(v sdk.ConsAddress, startHeight int64) []byte {
 	b := make([]byte, 8)
-	binary.LittleEndian.PutUint64(b, uint64(startHeight))
+	// this needs to be height + 1 because the slashing period for genesis validators starts at height -1
+	binary.LittleEndian.PutUint64(b, uint64(startHeight+1))
```

Comment:
More - there are * a lot * of comments on subtle changes throughout like "+1 because of XXX" /  "-1 because of something else" - I like what you've done by making `ValidatorDelay` its own variable - I'd like to see more variables if possible to describe some of these other `+1` / `-1` situations - then we could just have a single comment at the variable definition and the connections could be drawn more easily for someone attempting to understand the codebase

Reply:
I have changed these to `ValidatorDelay`. The reason I didn't really do that for everything originally was because I don't know if this kind of implementation makes sense for large values of `ValidatorDelay`, but this is probably clearer for now.